src/coreclr/jit/importer.cpp

```diff
@@ -8231,6 +8231,35 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
         }
         break;
 
+        case CORINFO_FIELD_STATIC_DATASEGMENT:
```

Comment:
This does look a lot like `getFieldAddress` with the second argument not set to a dummy value.

I realize it won't work as-is because the Jit assumes `getFieldAddress` will always return a direct address today (it should not be hard to fix that), it just does not feel great to have the duplicated functionality.

Reply:
The difference from `getFieldAddress` is that this gives RyuJIT a base+offset view instead of giving a single symbol that RyuJIT can't do much about (`getFieldAddress` returns the address with the base factored in).

It was the feedback from https://github.com/dotnet/corert/pull/5131#issuecomment-352765429 (my earlier attempt) that ended up going the `getFieldAddress` path.